tools/unused.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import b
+
+# @run_on_docker
+def register(command):
+    # Install help
+    command.help = "Creates a list of unused or commented out modules"
+
+
+# @run_on_docker
+def run(**kwargs):
+
+    source_path = b.cmake_cache["NUbots_SOURCE_DIR"]
+    roles_path = source_path + "/roles"
+    modules_path = source_path + "/modules"
+    # print(roles_path, modules_path)
+    existing_modules = {}
+    missing_modules = {}
+    commented_modules = {}
+    used_modules = []
+    unused_commented_modules = []
+
+    # a list of folders that could be within a module
+    skip_list = ["src", "data", "config", "tests", "report"]
+    skip_list = [s + os.sep for s in skip_list]
+
+    # find all the modules that are available
+    for root, dirs, files in os.walk(modules_path):
+        if len(files) != 0:
+            if not any(substring in (root + os.sep) for substring in skip_list):
+                existing_modules[
+                    root.replace(modules_path + os.sep, "")
+                ] = 0  # make sure we remove the base path from the module
+
+    # compare our roles to our existing modules
+    for root, dirs, files in os.walk(roles_path):
```

Comment:
`os.listdir` is plenty for the roles directory, we're not going to have roles in subdirectories